tests/system/src/test/java/test/javafx/scene/web/StraightLineTest.java

```diff
@@ -0,0 +1,176 @@
+        new Thread(() -> Application.launch(StraightLineTestApp.class, (String[])null)).start();
+
+        assertTrue("Timeout waiting for FX runtime to start", Util.await(launchLatch));
+    }
+
+    @AfterClass
+    public static void tearDownOnce() {
+        Platform.exit();
+    }
+
+    @Before
+    public void setupTestObjects() {
+        Platform.runLater(() -> {
+            webView = new WebView();
+            Scene scene = new Scene(webView,80,60);
+            straightLineTestApp.primaryStage.setScene(scene);
+            straightLineTestApp.primaryStage.show();
+        });
+    }
+
+    @Test public void testLine() {
+        final CountDownLatch webViewStateLatch = new CountDownLatch(1);
+
+        Util.runAndWait(() -> {
+            assertNotNull(webView);
+
+            webView.getEngine().getLoadWorker().stateProperty().
+                addListener((observable, oldValue, newValue) -> {
+                if (newValue == SUCCEEDED) {
+                    webView.requestFocus();
+                }
+            });
+
+            webView.focusedProperty().
+                addListener((observable, oldValue, newValue) -> {
+                if (newValue) {
+                    webViewStateLatch.countDown();
+                }
+            });
+
+            webView.getEngine().loadContent("<!DOCTYPE html>\n" +
+                    "<html>\n" +
+                    "<head>\n" +
+                    "<style>\n" +
+                    "div {\n" +
+                    "padding:0px;\n"+
+                    "  text-decoration: underline;\n" +
+                    "  text-decoration-thickness: 20px;\n" +
+                    "font-size: 16px; \n" +
+                    "}\n" +
+                    "</style>\n" +
+                    "</head>\n" +
+                    "<body>\n" +
+                    "<div>TEST</dv>\n" +
+                    "</body>\n" +
+                    "</html>");
+
+        });
+
+        assertTrue("Timeout when waiting for focus change ", Util.await(webViewStateLatch));
+        Util.sleep(1000);
+
+        Util.runAndWait(() -> {
+            WritableImage snapshot = straightLineTestApp.primaryStage.getScene().snapshot(null);
+            PixelReader pr = snapshot.getPixelReader();
+
+            int start_x = (int)webView.getEngine().executeScript("document.getElementsByTagName('div')[0].getBoundingClientRect().x");
+            int start_y = (int)webView.getEngine().executeScript("document.getElementsByTagName('div')[0].getBoundingClientRect().y");
+            int height = (int)webView.getEngine().executeScript("document.getElementsByTagName('div')[0].getBoundingClientRect().height");
+            int width = (int)webView.getEngine().executeScript("document.getElementsByTagName('div')[0].getBoundingClientRect().width");
+
+            // buttom start x position of underline ( startx + font size + thickness -1)
+            int line_start_x = start_x + height + 20 - 1;
```

Comment:
The bounding rect.x rect.y is top left corner, and line is being drawn below the bottom, so height and thickness need to be considered.

Reply:
int i = line_start_y; i < (line_start_y - 6); i++ , this meets the condition of sampling near the right edge

Reply:
You are mixing values that should affect `x` (width) with values that should affect `y` (height and thickness). Using height or thickness to adjust the `x` value or using width to adjust the `y` value is wrong.

One thing that might help make it clearer is to use `y` as the name of the loop variable rather than `i`. Another thing that might help is to define a `thickness` constant rather than using 20. Then it will be more obvious where you are mixing x and y.